src/index.test.ts

```diff
@@ -140,4 +140,44 @@ describe("Aracari", () => {
     const addresses = aracari.getAddressesForText("toucan");
     expect(addresses).toEqual(["0.21.0", "0.23.0"]);
   });
+  test("replaceText should not append nodes if no matches are found", () => {
+    const element = document.createElement("div");
+    const replacementOptions = { perserveWord: true };
+    // Taken from https://en.wikipedia.org/wiki/French_Polynesia#Culture
+    element.innerHTML = `<p><a href="/wiki/French_language" title="French language">French</a> is the only official language of French Polynesia.<sup id="cite_ref-35" class="reference"><a href="#cite_note-35">[31]</a></sup> An <a href="/wiki/Organic_law" title="Organic law">organic law</a> of 12 April 1996 states that "French is the official language, Tahitian and other Polynesian languages can be used." At the 2017 census, among the population whose age was 15 and older</p>`;
+    aracari = new Aracari(element);
+    // Replace the two that are in the sentence
+    aracari
+      .replaceText(
+        "language",
+        [document.createTextNode("foo")],
+        replacementOptions
+      )
+      .remap()
+      .replaceText(
+        "language",
+        [document.createTextNode("bar")],
+        replacementOptions
+      )
+      .remap();
+
+    expect(() => {
+      aracari
+        .replaceText(
+          "language",
+          [document.createTextNode("baz")],
+          replacementOptions
+        )
+        .remap()
+        .replaceText(
+          "language",
+          [document.createTextNode("qux")],
+          replacementOptions
+        )
+        .remap();
```

Comment:
Does this get called since we throw an error on that first replace text? I suppose as part of pre-processing before we hit aracari we should have a good understanding of what is in the sentence vs not though. Just want to make sure I'm following the logic here.

Reply:
Actually, in this instance, this does not run either remap or replace because of the throw. The thing I am emulating is the old behavior where essentially where replacements would double up. I made this test in a failing state where they would append the text to the end but I could pull it back to just the initial replace call. Think I will follow up with that.